graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -983,6 +1128,34 @@ int64_t Pick(
   }
 }
 
+template <typename PickedType>
+int64_t TemporalPick(
+    torch::Tensor seed_timestamp, torch::Tensor csc_indices,
+    int64_t seed_offset, int64_t offset, int64_t num_neighbors, int64_t fanout,
+    bool replace, const torch::TensorOptions& options,
+    const torch::optional<torch::Tensor>& probs_or_mask,
+    const torch::optional<torch::Tensor>& node_timestamp,
+    const torch::optional<torch::Tensor>& edge_timestamp,
+    PickedType* picked_data_ptr) {
+  auto mask = TemporalMask(
+      seed_timestamp[seed_offset].item<int64_t>(), csc_indices, probs_or_mask,
+      node_timestamp, edge_timestamp, {offset, offset + num_neighbors});
+  torch::Tensor masked_prob;
+  if (probs_or_mask.has_value()) {
+    masked_prob =
```

Comment:
If `prob_or_mask `is Bool, will `* ` cause error?

Reply:
prob_or_mask has been casted to float32 before this function.